internal/function/uast.go

```diff
@@ -40,10 +42,15 @@ func NewUAST(args ...sql.Expression) (sql.Expression, error) {
 		blob = args[0]
 		lang = args[1]
 		xpath = args[2]
+	case 4:
```

Comment:
I know it's a silly detail, but how about:
```
switch len(args) {
       default: return nil, sql.ErrInvalidArgumentNumber.New("1, 2, 3 or 4", len(args))
 	case 4:		 	
 		mode = args[3]
                fallthrough
 	case 3:		 	
 		xpath = args[2]
                fallthrough
 	case 2:		 	 		
 		lang = args[1]	
                fallthrough
 	case 1:
		blob = args[0]
}
```
less copy paste - up to you ;)

Reply:
I've left it was it was previously as `uast` now changed back to its original parameters. We can do a second round of refinement if we feel it's needed.